cirq/ops/op_tree.py

```diff
@@ -19,6 +19,7 @@
 
 from typing import Any, Iterable, Union, Callable
 
+from cirq.circuits.moment import Moment
```

Comment:
Ah. Break cycle here instead of elsewhere.

Reply:
I don't see a way to implement the `preserve_moments` boolean flag without using Moment in `isinstance` checks.

Reply:
That sounds reasonable.